flowfile_core/flowfile_core/routes/project.py

```
@@ -0,0 +1,338 @@
+    folder_path = validate_path_under_cwd(req.folder_path)
+    try:
+        project, result = project_sync.open_project(folder_path, current_user.id)
+    except FileNotFoundError as e:
+        raise HTTPException(404, str(e)) from e
+    except ValueError as e:
+        raise _import_error(e) from e
+    return OpenProjectOut(project=_payload(project), **result.to_dict())
+
+
+@router.get("/root", response_model=ProjectRootOut)
+def get_project_root(current_user=Depends(get_current_active_user)) -> ProjectRootOut:
```

Ruff (0.15.17)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)